__tests__/index.test.js

```diff
@@ -107,3 +107,49 @@ describe('GET /api/articles', () => {
         })
     });
 });
+
+describe("POST /api/articles/:article_id/comments", () => {
+    test('POST: 200 adds new comment for given article', () => {
+        const newComment = {
+            body: "10/10 enjoyed, will be reading this again.",
+            username: "butter_bridge"
+        };
+        return request(app)
+        .post("/api/articles/2/comments")
+        .send(newComment)
+        .expect(201)
+        .then(({body}) => {
+            expect(body.comment.comment_id).toBe(19);
+            expect(body.comment.body).toBe("10/10 enjoyed, will be reading this again.");
+            expect(body.comment.votes).toBe(0);
+            expect(body.comment.author).toBe("butter_bridge");
+            expect(body.comment.article_id).toBe(2);
+            expect(typeof body.comment.created_at).toBe("string");
+        }) 
+    });
+    test('POST: 400 sends an appropriate status and error message when given a non-existent article_id', () => {
+        const newComment = {
+            body: "10/10 enjoyed, will be reading this again.",
+            username: "butter_bridge"
+        };
+        return request(app)
+        .post("/api/articles/20/comments")
+        .send(newComment)
+        .expect(400)
+        .then((response) => {
+            expect(response.body.msg).toBe('Bad request');
+        });
+    });
+    test('POST:400 responds with an appropriate status and error message when provided with a bad comment (no username given)', () => {
+        const newComment = {
+            body: "10/10 enjoyed, will be reading this again."
+        };
+        return request(app)
+        .post("/api/articles/1/comments")
+        .send(newComment)
+        .expect(400)
+        .then((response) => {
+            expect(response.body.msg).toBe('Bad request');
+        });
+    });
```

Comment:
We should add a test for when we are passing an username that is not found(404), a newComment that is missing either the body or the username (400), and maybe one with additional unrelated properties (200) like {
            body: "10/10 enjoyed, will be reading this again.",
            username: "butter_bridge",
            fruit: "banana"
        };

Reply:
the error for passing an invalid username(username not found) is coming up as a 400 error (has error code), is this ok? is there a way to manually change a psql error to a custom error?

Reply:
"maybe one with additional unrelated properties (200) like {
body: "10/10 enjoyed, will be reading this again.",
username: "butter_bridge",
fruit: "banana" <<<< i am unsure if this is an error test or a happy test? when i tried it , it didn't throw an error, not sure how to make it do this?